docs/guides/latest/workloads/deployment.md

```diff
@@ -0,0 +1,435 @@
+$ kubectl create secret generic -n demo local-secret \
+    --from-file=./RESTIC_PASSWORD
+secret/local-secret created
+```
+
+Verify that the secret has been created successfully,
+
+```console
+$ kubectl get secret -n demo local-secret -o yaml
+```
+
+**Create Repository:**
+
+Now, create a `Respository` using this secret. 
+
+Below is the YAML of `Repository` crd we are going to create,
+
+```console
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: repo-pvc
+  namespace: demo
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: local-repo
+  namespace: demo
+spec:
+  backend:
+    local:
+      mountPath: /safe/data
+      persistentVolumeClaim:
+        claimName: repo-pvc
+    storageSecretName: local-secret
+```
+
+Let's create the Repository we have shown above,
+
+```console
+$ kubectl apply -f ./local_repository.yaml 
+persistentvolumeclaim/repo-pvc created
+repository.stash.appscode.com/local-repo created
+```
+
+Now, we are ready to backup our volumes to our desired backend.
+
+### Backup
+
+We have to create a `BackupConfiguration` crd. Then Stash will create a `CronJob` to take periodically backup of the PVC of `stash-demo` Deployment.
+
+**Create BackupConfiguration:**
+
+Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: deployment-backup
+  namespace: demo
+spec:
+  repository:
+    name: local-repo
+  schedule: "*/1 * * * *"
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-demo
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+    directories:
+    - /source/data
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.repository` refers to the `laocal-repo` local Backend.
+- `spec.schedule` is a cron expression indicates that `BackupSession` will be created at 1 minute interval.
+- `spec.target.ref` refers to the target workload that was created for `stash-demo` Deploymnet.
+
+Let's create the `BackupConfiguration` crd we have shown above,
+
+```console
+ $ kubectl apply -f ./backupconfiguration.yaml
+backupconfiguration.stash.appscode.com/deployment-backup created
+```
+
+**Verify CronJob:**
+
+If everything goes well, Stash will create a `CronJob` with the schedule specified in `spec.schedule` field of `BackupConfiguration` crd.
+
+Verify that the `CronJob` has been created using the following command,
+
+```console
+$ kubectl get cronjob -n demo
+NAME                SCHEDULE      SUSPEND   ACTIVE   LAST SCHEDULE   AGE
+deployment-backup   */1 * * * *   False     0        35s             64s
+```
+
+**Wait for BackupSession:**
+
+The `deployment-backup` CronJob will trigger a backup on each schedule by creating a `BackpSession` crd.
```

Comment:
Add:

The sidecar container will watches for the `BackupSession` crd. Once it found one, it will take backup immediately.